src/Orleans/Streams/Internal/StreamConsumer.cs

```diff
@@ -172,6 +172,32 @@ public Task<StreamSubscriptionHandle<T>> SubscribeAsync(IAsyncObserver<T> observ
                                   .ToList<StreamSubscriptionHandle<T>>();
         }
 
+        public async Task Cleanup()
+        {
+            if (logger.IsVerbose) logger.Verbose("Cleanup() called");
+            if (myExtension == null)
+                return;
+
+            var allHandles = myExtension.GetAllStreamHandles<T>();
```

Comment:
Does it make sense to add some sort of RemoveAllSubscriptions call to the pubsub system.  In the most common case for us (single subscription) this is still two grain calls.  In the case of multiple subscriptions this could be multiple grain calls, each triggering a table storage write.